verilog/CST/declaration_test.cc

```diff
@@ -682,5 +682,139 @@ TEST(FindAllDataDeclarationTest, FindDataDeclarationParameters) {
   }
 }
 
+TEST(GetVariableDeclarationAssign,
+     FindUnpackedDimensionOfVariableDeclarationAssign) {
+  constexpr int kTag = 1;  // value doesn't matter
+  const SyntaxTreeSearchTestCase kTestCases[] = {
+      {""},
+      {"module m;\nendmodule\n"},
+      {"class class_c;\nendclass\nmodule m;\nclass_c c = new();\nendmodule"},
+      {"package pkg;\nint x",
+       {kTag, "[k:y]"},
+       " = 4, y ",
+       {kTag, "[k:y]"},
+       " = 4;\nlogic k ",
+       {kTag, "[k:y]"},
+       " = fun_call();\nendpackage"},
+      {"class cls;\n int x ",
+       {kTag, "[k:y]"},
+       " = 4, y ",
+       {kTag, "[k:y]"},
+       " = 4;\nlogic k ",
+       {kTag, "[k:y]"},
+       " = fun_call();\nendclass"},
+  };
```

Comment:
Suggest a few test cases with both packed/unpacked dimensions so we know these functions select the correct one.  (Same below)